borsh/src/de/mod.rs

```diff
@@ -370,6 +370,28 @@ impl BorshDeserialize for String {
     }
 }
 
+#[cfg(feature = "ascii")]
+impl BorshDeserialize for ascii::AsciiString {
+    #[inline]
+    fn deserialize_reader<R: Read>(reader: &mut R) -> Result<Self> {
+        let bytes = Vec::<u8>::deserialize_reader(reader)?;
+        ascii::AsciiString::from_ascii(bytes)
+            .map_err(|err| Error::new(ErrorKind::InvalidData, err.to_string()))
+    }
+}
+
+#[cfg(feature = "ascii")]
+#[test]
+fn test_ascii() {
```

Comment:
please change logic and naming of tests:
`test_ascii_string_to_ascii_string_roundtrip`
`test_ascii_str_to_ascii_string_roundtrip`
And 2 separate tests for `ascii`/`std`-`Strings` interoperability (which are optional):
`test_str_to_ascii_string_maybe_roundtrip`
`test_ascii_str_to_string_roundtrip`.

Reply:
I will definitely not do that.